x/dex/keeper/msgserver/msg_server_place_orders.go

```diff
@@ -73,11 +69,23 @@ func (k msgServer) PlaceOrders(goCtx context.Context, msg *types.MsgPlaceOrders)
 		order.Id = nextID
 		order.Account = msg.Creator
 		order.ContractAddr = msg.GetContractAddr()
-		dexutils.GetMemState(ctx.Context()).GetBlockOrders(ctx, typesutils.ContractAddress(msg.GetContractAddr()), pairStr).Add(order)
+		ordersToAdd[pairStr] = order
 		idsInResp = append(idsInResp, nextID)
 		nextID++
 	}
+	// setting bumped order id BEFORE adding orders to cache, so that if the latter fails, at least we wouldn't have
```

Comment:
hm, thinking about this more, how does duplicated orderId happen again? like if the order fails, it theoretically should never show up so we should be able to reuse the order.
Or is the idea here just to make it more resilient b/c there's the unknown ongoing issue?